synapse/config/appservice.py

```diff
@@ -30,12 +31,12 @@
 class AppServiceConfig(Config):
     section = "appservice"
 
-    def read_config(self, config, **kwargs):
```

Comment:
`config: Dict[str, Any]`?

kwargs seems to be `config_dir_path: str` and `data_dir_path: str` but I don't think these are frequently used. But let me not scope creep...

Reply:
I have a separate branch built on this which adds type hints to `read_config`.

Reply:
Ahh, my mistake.

Reply:
No problem! I should have included it in the description. :) There's a lot to these classes, so trying to break it up logically...but maybe only logically in my head. :)